shared/src/shared.rs

```diff
@@ -146,6 +166,67 @@ impl Shared {
         }
     }
 
+    pub fn spawn_freeze(&self) -> FreezerClose {
+        let (signal_sender, signal_receiver) =
+            crossbeam_channel::bounded::<()>(service::SIGNAL_CHANNEL_SIZE);
+        let shared = self.clone();
+        let thread = thread::Builder::new()
+            .spawn(move || loop {
+                match signal_receiver.recv_timeout(FREEZER_INTERVAL) {
+                    Err(_) => {
+                        if let Err(e) = shared.freeze() {
```

Comment:
Freezer continues on error, so it must be able to recover from errors automatically.